src/software/ai/hl/stp/play/corner_kick_play_test.cpp

```diff
@@ -74,3 +74,99 @@ TEST_F(CornerKickPlayTest, test_corner_kick_play_top_right)
             terminating_validation_functions, non_terminating_validation_functions,
             Duration::fromSeconds(10));
 }
+
+class CornerKickPlayInvariantAndIsApplicableTest
+    : public SimulatedPlayTestFixture,
+      public ::testing::WithParamInterface<
+          std::tuple<TeamSide, RefereeCommand, RefereeCommand, Point, Vector, bool, bool>>
+{
+   protected:
+    Field field = Field::createSSLDivisionBField();
+};
+
+TEST_P(CornerKickPlayInvariantAndIsApplicableTest, test_invariant_and_is_applicable)
+{
+    // Set world variables
+    auto play_config = std::make_shared<ThunderbotsConfig>()->getPlayConfig();
+
+    auto world = ::TestUtil::createBlankTestingWorld();
+
+    auto corner_kick_play = CornerKickPlay(play_config);
+
+    // -------------------------------------------------------------------------
+    // Check if isApplicable() and invariantHolds() for various world states and ball
+    // positions
+
+    // initialize world state from parameters
+    world.setTeamWithPossession(std::get<0>(GetParam()));
+    world.updateGameState(
+        ::TestUtil::createGameState(std::get<1>(GetParam()), std::get<2>(GetParam())));
+
+    // initialize ball from parameters
+    BallState ball_state_in_corner(std::get<3>(GetParam()), std::get<4>(GetParam()));
```

Comment:
`std::get<4>(GetParam())` isn't necessary since it's always `Vector (0,0)`, so remove that 4th param and do
```suggestion
    BallState ball_state_in_corner(std::get<3>(GetParam()), Vector(0,0));
```

Reply:
Should be updated now.